util.rb

```diff
@@ -0,0 +1,61 @@
+require 'sinatra/base'
+require_relative 'environment'
+require 'time'
+
+# Util is a class of utility methods
+class Util
+  def self.max_value(data)
+    # Return the maximum value field from a list of hashes
+    # an empty list should return 0
+    return 0 if data.empty?
+    max = data[0][:value]
+    data.each do |line|
+      max = line[:value] if line[:value] > max
+    end
+    max # return
+  end # maxValue
+
+  def self.min_value(data)
+    # Return the minimum Value field in a list of hashes
+    # an empty list should return 0
+    return 0 if data.empty?
+    min = data[0][:value]
+    data.each do |line|
+      min = line[:value] if line[:value] < min
+    end
+    min # return
+  end # minValue
+
+  def self.avg_value(data)
+    # Return the avg (mean) Value field in a list of hashes
+    # an empty list should return 0
+    return 0 if data.empty?
+    total = 0
+    data.each do |line|
+      total += line[:value]
+    end
+    total /= data.length
+  end # avgValue
+
+  def self.days_in_range(data)
+    # Return number of days in range of data
+    # an empty range includes 0 days
+    return 0 if data.empty?
+    earliest = data[0][:created]
+    latest = data[-1][:created]
+    now = Time.now() # dates cannot be in the future
+    data.each do |line|
+      l = line[:created]
+      # dates must be in a Time format
+      return nil if l.class != Time
+      # dates cannot be in the future
+      return nil if l > now
+      earliest = l if l < earliest
+      latest = l if l > latest
+    end
+    # latest-earliest is number of seconds between two dates
+    # there are 60*60*24 seconds every day (leap years? DST?)
+    # round up to whole days?  Maybe not?
+    ((latest - earliest) / (60 * 60 * 24)).ceil
```

Comment:
Thanks for the feedback @mathuin.

I've been thinking about it and I suspect this should work fine with DST and leap years, because the time stamps are in UTC and (should) play nice  with the additional day of a leap year.

I'm not sure how to test DST, but I can definitely add a leap year test by choosing days on either side of a known leap year.  I'll add that test case when I get in tomorrow.

I think you're probably right, a float can be rounded later in the program, but returning an int in the method reduces precision.  I'll check with @ramereth and @Kennric to confirm.